src/core/ext/transport/chttp2/client/insecure/channel_create.cc

```diff
@@ -109,3 +109,27 @@ grpc_channel* grpc_insecure_channel_create(const char* target,
                                   target, GRPC_STATUS_INTERNAL,
                                   "Failed to create client channel");
 }
+
+/* The same as grpc_insecure_channel_create expect for:
+ *   - doesn't create a new ExecCtx
+ *   - doesn't call grpc_init or grpc_shutdown internally */
+grpc_channel* grpc_insecure_channel_create_internal(
```

Comment:
Instead of duplicating this code, how about just having the ALTS handshaker set the channel arg before it calls `grpc_insecure_channel_create()`?

I don't actually care about the `ExecCtx` stacking.  The code can currently handle that, and we're trying to make `ExecCtx` go away completely in the long run anyway.

Reply:
> I don't actually care about the ExecCtx stacking. The code can currently handle that, and we're trying to make ExecCtx go away completely in the long run anyway.

A little late response but not sure I understand here BTW. Speaking generally, my view is that `ExecCtx` stacking is error prone due to re-entrant locking risk. https://github.com/grpc/grpc/pull/20596 actually tries hard to avoid it.

Reply:
@vjpai or @yashykt can comment in more detail, but it's my understanding that the ExecCtx code automatically detects when this stacking is happening and basically shunts all of the queued closures to the "root" ExecCtx instance, so there should not be any risk of deadlock.  At this point, I'd rather be putting effort into making ExecCtx go away instead of jumping through hoops trying to avoid stacking.

Reply:
Currently, ExecCtx can get stacked on top of one another. Closures are always queued to the most recent/top ExecCtx.

Also, it doesn't look like ExecCtx is gonna go away :)

Reply:
I think we need to put effort into either eliminating the stacking functionality or removing ExecCtx altogether.  We'll have to do some experiments to see which way we'll go.